### PR TITLE
Rename ddprof-ffi crate to datadog-profiling-ffi

### PR DESCRIPTION
**Pull request by morrisonlevi**
# What does this PR do?

- This renames the `ddprof-ffi` crate to `datadog-profiling-ffi`. It renames the folder accordingly.
- Format cmake files and add Secur32 and Ncrypt libs on Win. Move libs into main cmake folder (instead of the examples) so anyone using the cmake files get these linked.
- Clean up ffi examples a bit.
- Bumps version to `0.8.0-alpha.1`.

# Motivation

Before going 1.0, we wanted to do a pass on naming.

# Additional Notes

Originally, this PR moved the ffi crate into the profiling crate as a feature-enabled module. Although I liked this better, the file sizes increased 50-70% because of an LTO bug in rust: https://github.com/rust-lang/rust/issues/51009. 

# How to test the change?

The pkg-config and cmake files names changed, as did the name of the packages. Once those are fixed, upgrading should be pretty seamless.

### CODE REVIEW

**Review by pawelchcki — APPROVED** [review]
Looks good to me :)

File: cmake/DatadogConfig.cmake.in [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  set(Datadog_INCLUDE_DIRS ${Datadog_INCLUDE_DIR})
+  set(Datadog_LIBRARIES ${Datadog_LIBRARY} "@Datadog_LIBRARIES@")
+  mark_as_advanced(Datadog_ROOT Datadog_LIBRARY Datadog_INCLUDE_DIR)
+
+  add_library(datadog_profiling INTERFACE)
+  target_include_directories(datadog_profiling
+                             INTERFACE ${Datadog_INCLUDE_DIRS})
+  target_link_libraries(datadog_profiling INTERFACE ${Datadog_LIBRARIES})
+  target_compile_features(datadog_profiling INTERFACE c_std_11)
+  target_compile_definitions(datadog_profiling INTERFACE DATADOG_PROFILING_FFI)
```
Comment (nsavoire) [inline cmake/DatadogConfig.cmake.in:33]: `DATADOG_PROFILING_FFI` seems unused

File: profiling-ffi/datadog_profiling-static.pc.in [hunk 2]
```diff
@@ -8,10 +8,10 @@ exec_prefix=${prefix}
...
+Name: datadog_profiling
 Description: Contains common code used to implement Datadog's Continuous Profilers. (Static linking variant)
-Version: @DDProf_FFI_VERSION@
+Version: @Datadog_VERSION@
 Requires:
-Libs: -L${libdir} ${libdir}/libddprof_ffi.a @DDProf_FFI_LIBRARIES@
+Libs: -L${libdir} ${libdir}/libdatadog_profiling.a @Datadog_LIBRARIES@
 Libs.private:
-Cflags: -I${includedir}
+Cflags: -I${includedir} -DDATADOG_PROFILING_FFI
```
Comment (nsavoire) [inline profiling-ffi/datadog_profiling-static.pc.in:17]: `DATADOG_PROFILING_FFI` seems unused

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@ variables:
   # These are gitlab variables so that it's easier to do a manual deploy
   # If these are set witih value and description, then it gives you UI elements
   DOWNSTREAM_BRANCH:
-    value: "main"
+    value: "levi/bye-ddprof-ffi"
```
Comment (nsavoire) [inline .gitlab-ci.yml:5]: Might need to remove this before merging I guess.

**Review by nsavoire — APPROVED** [review]
LGTM, thanks for the cleanup !

File: cmake/DatadogConfig.cmake.in (hunk 1 above)
Comment (morrisonlevi) [reply cmake/DatadogConfig.cmake.in:33]: That's right, it's leftover from pushing it into the profiling crate. Good catch!

File: .gitlab-ci.yml (hunk 3 above)
Comment (morrisonlevi) [reply .gitlab-ci.yml:5]: Just merged the branch; will push the change now. Thanks!

**Review by ivoanjo — COMMENTED** [review]
Late review, but it LGTM 👍 

I'm soft-concerned about the file size increase, but will measure and report in the channel if there's any concerns separately :)

**Pull request merged**